### PR TITLE
LPS-83801 Cache class restriction computation

### PR DESCRIPTION
**Pull request by csierra**
could you please run performance tests against this one to see if it is any better?

thanks!

/cc @topolik

### CODE REVIEW

**Comment by slnn** [comment]
Hi @csierra and @shuyangzhou 

Here are test results for this pull, no noticeable performance regression appeared.
Login:

Portal Version | Git Id | Session Count | Mean Time of Login | Mean Time of Logout
-- | -- | -- | -- | --
BaseLine | feb06f5 | 48471 | 276 | 84.4
PR-324 | - | 48764 | 280 | 85.3

WebContent:

Portal Version | Git Id | Session Count | Mean Time of Login | Mean Time of View Page
-- | -- | -- | -- | --
BaseLine | feb06f5 | 61531 | 603 | 217
PR-324 | - | 63299 | 549 | 191

Wiki

Portal Version | Git Id | Session Count | Mean Time of Login | Mean Time of Add Wiki Page
-- | -- | -- | -- | --
BaseLine | feb06f5 | 15580 | 447 | 642
PR-324 | - | 15151 | 496 | 681

AssetPublisher:

Portal Version | Git Id | Session Count | Mean Time of Login | Mean Time of View Page with Filter
-- | -- | -- | -- | --
BaseLine | feb06f5 | 24222 | 478 | 208
PR-324 | - | 24132 | 468 | 209

Blog:

Portal Version | Git Id | Session Count | Mean Time of Login | Mean Time of Add Blog Entry
-- | -- | -- | -- | --
BaseLine | feb06f5 | 13174 | 150 | 251
PR-324 | - | 13132 | 205 | 287

MessageBoard:

Portal Version | Git Id | Session Count | Mean Time of Login | Mean Time of Add MB Thread
-- | -- | -- | -- | --
BaseLine | feb06f5 | 15866 | 43.6 | 79.7
PR-324 | - | 15609 | 22.9 | 61.1

DocumentLibrary:

Portal Version | Git Id | Session Count | Mean Time of Login | Mean Time of Add DL File
-- | -- | -- | -- | --
BaseLine | feb06f5 | 26207 | 321 | 385
PR-324 | - | 25832 | 237 | 324

**Comment by csierra** [comment]
thx!
sent to https://github.com/topolik/liferay-portal/pull/600

File: modules/apps/portal-template/portal-template-freemarker/src/main/java/com/liferay/portal/template/freemarker/internal/LiferayObjectWrapper.java [hunk 1]
```diff
@@ -234,6 +211,64 @@ else if (object instanceof ResourceBundle) {
...
+	private void _checkClassIsRestricted(Class<?> clazz)
+		throws TemplateModelException {
+
+		String className = clazz.getName();
+
+		ClassRestrictionInformation classRestrictionInformation =
+			_classRestrictionInformations.computeIfAbsent(
+				className,
+				a -> {
+					if (_allowedClassNames.contains(className)) {
```
Comment (tinatian) [inline modules/apps/portal-template/portal-template-freemarker/src/main/java/com/liferay/portal/template/freemarker/internal/LiferayObjectWrapper.java]: @csierra This checking does make sense to me. If the classes is in allowed list, you should not block it.

Comment (csierra) [reply modules/apps/portal-template/portal-template-freemarker/src/main/java/com/liferay/portal/template/freemarker/internal/LiferayObjectWrapper.java]: hey @tinatian you are absolutely right... it looks I turned the test around when I cached that
fixing...
thx!

**Pull request closed without merging**